tests/phpunit/Compat/WC_Stripe_Subscriptions_Helper_Test.php

```diff
@@ -43,4 +45,122 @@ public function test_is_manual_renewal_enabled() {
 		// If the logic changes, this test should be updated accordingly.
 		$this->assertFalse( WC_Stripe_Subscriptions_Helper::is_manual_renewal_enabled() );
 	}
+
+	/**
+	 * Test for `get_detached_subscriptions`.
+	 *
+	 * @return void
+	 */
+	public function test_get_detached_subscriptions() {
+		$subscription_id = 1;
+		$customer_id     = 'cus_123';
+		$source_id       = 'src_123';
+
+		$subscription = new WC_Subscription();
+		$subscription->set_id( $subscription_id );
+		$subscription->set_status( 'active' );
+		$subscription->set_time( 'next_payment_date', strtotime( '+1 week' ) );
+		$subscription->save();
+
+		$subscription->update_meta_data( '_stripe_customer_id', $customer_id );
+		$subscription->update_meta_data( '_stripe_source_id', $source_id );
+		$subscription->save_meta_data();
+
+		WC_Subscriptions_Helpers::$wcs_get_subscriptions = [ $subscription ];
+
+		// Mock response from Stripe API.
+		$test_request = function () {
+			return [
+				'response' => 200,
+				'headers'  => [ 'Content-Type' => 'application/json' ],
+				'body'     => wp_json_encode(
+					[
+						'customer' => null,
+					]
+				),
+			];
+		};
+
+		add_filter( 'pre_http_request', $test_request, 10, 3 );
+
+		// Mock the change payment method URL.
+		$mocked_payment_method_url = 'https://example.com/my-account/subscription-payment-method/' . $subscription_id;
+		add_filter(
+			'wcs_get_change_payment_method_url',
+			function () use ( $mocked_payment_method_url ) {
+				return $mocked_payment_method_url;
+			},
+			10,
+			2
+		);
+
+		$expected = [
+			[
+				'id'                        => $subscription_id,
+				'customer_id'               => $customer_id,
+				'change_payment_method_url' => $mocked_payment_method_url,
+			],
+		];
+		$this->assertEquals( $expected, WC_Stripe_Subscriptions_Helper::get_detached_subscriptions() );
+
+		remove_filter( 'pre_http_request', $test_request, 10, 3 );
```

Comment:
To avoid failed assertions leaving the `pre_http_request` filter set up, might it make sense to handle the cleanup before the assertion?
```suggestion
		$result = WC_Stripe_Subscriptions_Helper::get_detached_subscriptions();

		// Clean up before we run any assertions
		remove_filter( 'pre_http_request', $test_request, 10, 3 );

		$this->assertEquals( $expected, $result );

```